### ISSUE TITLE
0.2.7 CPU Memory Leak - MPT-7B tensor_parallel_size=4

### ISSUE THREAD

**Issue by tosterberg** [vllm-project/vllm#2793]
I am running `mosaicml/mpt-7b` on `vllm==0.2.7`, with `cuda==12.1` on a `g5.12xlarge` instance and after continuous use I get an Out of Memory error from my CPU's memory. Below is my script for minimum reproducibility. 

```
from vllm import LLM
import psutil

llm = LLM("mosaicml/mpt-7b", trust_remote_code=True, tensor_parallel_size=4)

for i in range(0, 100000):
    output = llm.generate("San Franciso is a")
    if i % 1000 == 0:
        cpu_percent = psutil.cpu_percent()
        memory_percent = psutil.virtual_memory().percent
        with open("cpu_memory.log","a") as f:
            f.write(f"Iteration number: {i}\n")
            f.write(f"CPU utilization: {cpu_percent}%\n")
            f.write(f"Memory utilization: {memory_percent}%\n")
            f.write("=========================================\n")

print("Done!")
```

Running the above script for a while shows a starting CPU Memory usage % of 14.5% and by 20,000 inferences 21%. Previously, `vllm==0.2.1`, we were able to run our endpoint continuously without an OOM causing a restart but now it will OOM every 8 hours or so.

**Comment by tosterberg** [comment]
I am testing to see if this issue remains for `vllm==0.3.0`.

**Comment by Alienfeel** [comment]
same +1, when serving mistral-7B

**Comment by SamComber** [comment]
+1 here but no tensor parallelism, saw deployment of mistral-7b cause GPU memory to leak across a number of hours on a A10G running `v0.3.0`. Any help here guys ? @WoosukKwon (amazing library btw)

21646MiB / 22731MiB 
22056MiB / 22731MiB 
22264MiB / 22731MiB

**Comment by hanzhi713** [comment]
@SamComber @Alienfeel @tosterberg  You need to add `enforce_eager=True`. Currently, pytorch NCCL with cuda graph have CPU memory leak.

**Comment by SamComber** [comment]
Thanks @hanzhi713 ! Is there any PR / discussion somewhere on this so we can track the issue?

**Comment by hanzhi713** [comment]
There's a pending PR that uses cupy for allreduce and may solve the memory leak problem

https://github.com/vllm-project/vllm/pull/2811

**Comment by WoosukKwon** [comment]
Should be fixed by #2811

**Comment by WoosukKwon** [comment]
@tosterberg Thanks for providing the reproducible example. We just merged the fix. Please try the main branch!
